php/example_code/s3/S3Service.php

```diff
@@ -0,0 +1,238 @@
+            if ($this->verbose) {
+                echo "Created the bucket named: $bucketName.\n";
+            }
+        } catch (AwsException $exception) {
+            if ($this->verbose) {
+                echo "Failed to create $bucketName with error: {$exception->getMessage()}\n";
+                echo "Please fix error with bucket creation before continuing.";
+            }
+            throw $exception;
+        }
+    }
+
+    public function putObject(string $bucketName, string $key, array $args = [])
+    {
+        $parameters = array_merge(['Bucket' => $bucketName, 'Key' => $key], $args);
+        try {
+            $this->client->putObject($parameters);
+            if ($this->verbose) {
+                echo "Uploaded the object named: $key to the bucket named: $bucketName.\n";
+            }
+        } catch (AwsException $exception) {
+            if ($this->verbose) {
+                echo "Failed to create $key in $bucketName with error: {$exception->getMessage()}\n";
+                echo "Please fix error with object uploading before continuing.";
+            }
+            throw $exception;
+        }
+    }
+
+    public function getObject(string $bucketName, string $key, array $args = []): Result
+    {
+        $parameters = array_merge(['Bucket' => $bucketName, 'Key' => $key], $args);
+        try {
+            $object = $this->client->getObject($parameters);
+            if ($this->verbose) {
+                echo "Downloaded the object named: $key to the bucket named: $bucketName.\n";
+            }
+        } catch (AwsException $exception) {
+            if ($this->verbose) {
+                echo "Failed to download $key from $bucketName with error: {$exception->getMessage()}\n";
+                echo "Please fix error with object downloading before continuing.";
+            }
+            throw $exception;
+        }
+        return $object;
+    }
+
+    public function copyObject($bucketName, $key, $copySource, array $args = [])
+    {
+        $parameters = array_merge(['Bucket' => $bucketName, 'Key' => $key, "CopySource" => $copySource], $args);
+        try {
+            $this->client->copyObject($parameters);
+            if ($this->verbose) {
+                echo "Copied the object from: $copySource in $bucketName to: $key.\n";
+            }
+        } catch (AwsException $exception) {
+            if ($this->verbose) {
+                echo "Failed to copy $copySource in $bucketName with error: {$exception->getMessage()}\n";
+                echo "Please fix error with object copying before continuing.";
+            }
+            throw $exception;
+        }
+    }
+
+    public function listObjects(string $bucketName, $start = 0, $max = 1000, array $args = [])
+    {
+        $parameters = array_merge(['Bucket' => $bucketName, 'Marker' => $start, "MaxKeys" => $max], $args);
+        try {
+            $objects = $this->client->listObjects($parameters);
```

Comment:
I think you're supposed to use listObjectsV2 now.

Reply:
Updated.